src/cdk-experimental/menu/menu-item-trigger.ts

```diff
@@ -175,16 +179,31 @@ export class CdkMenuItemTrigger implements OnDestroy {
   }
 
   /**
-   * If there are existing open menus and this menu is not open, close sibling menus and open
-   * this one.
+   * Subscribe to the mouseenter events and close any sibling menu items if this element is moused
+   * into.
    */
-  _toggleOnMouseEnter() {
-    const menuStack = this._getMenuStack();
-    const isSiblingMenuOpen = !menuStack.isEmpty() && !this.isMenuOpen();
-    if (isSiblingMenuOpen) {
-      this._closeSiblingTriggers();
-      this.openMenu();
-    }
+  private _subscribeToMouseEnter() {
+    // Closes any sibling menu items and opens the menu associated with this trigger.
+    const toggleMenus = () =>
+      this._ngZone.run(() => {
+        this._closeSiblingTriggers();
+        this.openMenu();
+      });
+
+    this._ngZone.runOutsideAngular(() => {
+      fromEvent(this._elementRef.nativeElement, 'mouseenter')
+        .pipe(
+          filter(() => !this._getMenuStack()?.isEmpty() && !this.isMenuOpen()),
+          takeUntil(this._destroyed)
+        )
+        .subscribe(() => {
+          if (this._menuAim) {
+            this._menuAim.toggle(toggleMenus);
+          } else {
+            toggleMenus();
+          }
+        });
+    });
```

Comment:
Here's my thinking:
* It feels odd to me for `MenuAim` to be responsible to actually performing the close rather than just providing back the signal that the cursor is moving a certain way. Some like this (below) would keep that logic in the menu trigger and keep the menu aim more tightly focused 

```ts
    this._ngZone.runOutsideAngular(() => {
      const mouseenters =
          fromEvent(this._elementRef.nativeElement, 'mouseenter').pipe(
            filter(() => !this._getMenuStack()?.isEmpty() && !this.isMenuOpen()),
            takeUntil(this._destroyed),
          );

      // In a naive menu, any sbling triggers will have their submenus closed
      // open moving the mouse onto this trigger. However, we want to avoid this
      // if the user is moving their mouse towards the submenu.
      mouseenters.subscribe(() => {
        if (!this?.menuAim.isMovingTowardsSubmenu()) {
          this._closeSiblingTriggersWithinZone();
        } else {
          this._schedulePotentialSiblingMenuClose();
        }
      });
    });
```

Reply:
`_schedulePotentialSiblingMenuClose` would contain the delay logic (set timeout)? The one issue could be that we want to ignore any old setTimeout calls that may execute, hence we track the timeout id in the MenuAim service. This could be done using a service which is shared across all menu items in a menu and tracks the latest setTimeout call but that didn't seem ideal.

Reply:
That's a good point- the scheduling code is also only necessary if `menuAim` exists, so it would be better off there anyway. So maybe something like the above except it's `menuAim.schedulePotentialSiblingClose`

Reply:
Since it does more than close the sibling (it also possibly opens its own menu) how about something like `scheduleStackMutation` (since the top most menu item in the stack may pop off and a new item may be added)?

Reply:
I see. I think for now I'm fine with leaving this as-is, then.